File: serv/test/cgi-bin/hello.php
<?php
$method = $_SERVER['REQUEST_METHOD'] ?? 'GET';
$query  = $_SERVER['QUERY_STRING'] ?? '';
$name   = $_GET['name'] ?? 'world';

$body  = "<html><body>";
$body .= "<h1>Hello from PHP, " . htmlspecialchars($name) . "!</h1>";
$body .= "<p>METHOD: " . $method . "</p>";
$body .= "<p>QUERY_STRING: " . $query . "</p>";
$body .= "<p>SERVER_PORT: " . ($_SERVER['SERVER_PORT'] ?? '') . "</p>";

if ($method === 'POST') {
    $post_data = file_get_contents('php://stdin');
    $body .= "<p>POST body: " . htmlspecialchars($post_data) . "</p>";
}

$body .= "</body></html>";

header("Content-Type: text/html");
header("Content-Length: " . strlen($body));
echo $body;
?>

File: serv/test/cgi-bin/hello.py
#!/usr/bin/env python3
import os
import sys

method = os.environ.get("REQUEST_METHOD", "GET")
query  = os.environ.get("QUERY_STRING", "")

body = "<html><body>"
body += "<h1>CGI works</h1>"
body += "<p>METHOD: " + method + "</p>"
body += "<p>QUERY_STRING: " + query + "</p>"

if method == "POST":
    length = int(os.environ.get("CONTENT_LENGTH", 0) or 0)
    post_data = sys.stdin.read(length) if length else ""
    body += "<p>POST body: " + post_data + "</p>"

body += "</body></html>"

print("Content-Type: text/html")
print("Content-Length: " + str(len(body)))
print("")
print(body, end="")
